### ISSUE TITLE
Exponential Height Fog

### ISSUE THREAD

**Issue by Kliaxe** [turanszkij/WickedEngine#274]
HI turanszkij!

I've been playing around with an extension to the existing fog implemented into the engine. This would allow the user to create more unique scenes and additionally bring more realism with atmospheric like properties. Thought this might be an interesting addition to the engine.
Would this benefit the project?

Here are some screenshots, of the fog in engine:
![sc_02-07-2021 22-04-10](https://user-images.githubusercontent.com/16865975/124331009-de848d80-db8e-11eb-9050-e5744bff1b1a.jpg)
![sc_02-07-2021 22-14-56](https://user-images.githubusercontent.com/16865975/124331013-e04e5100-db8e-11eb-88da-483c7832bd80.jpg)

**Comment by turanszkij** [comment]
This would be great, I wanted to have this too. I tried to do something like this, but couldn't find a nice looking result. How does it affect the current fog parameters? It's important that the current fog parameters would still work the same way as they are already saved in the models. The current fog height is only used by the sky it seems, so that is a bit of an issue. I'd be interested to see your implementation.

Also, is that city model available freely somewhere?

**Comment by Kliaxe** [comment]
I'm glad you like it! The fog requires parameters for a start and end height. In this implementation I used the existing fog height as the start height, and added another parameter to adjust the end height. I was also a bit sceptical about using the existing one since other scenes uses it for the sky, so I completely agree we'd have to do something about it. I'll create the PR soon.

I city model on the screenshot was something I found on sketchfab, but note it's a bit skewed: https://sketchfab.com/3d-models/city-model-7506a2f9d3144fbdaecb713e5a7ff545
